src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -581,7 +629,7 @@ def __init__(self, config):
         self.wpe = nn.Embedding(config.max_position_embeddings, self.embed_dim)
 
         self.drop = nn.Dropout(config.embd_pdrop)
-        self.h = nn.ModuleList([GPT2Block(config) for _ in range(config.num_hidden_layers)])
+        self.h = nn.ModuleList([GPT2Block(config, layer_idx=i + 1) for i in range(config.num_hidden_layers)])
```

Comment:
Can we have `layer_idx=i`? It's more in line how it's done for other models: https://github.com/huggingface/transformers/blob/3fbb55c75779824aacfc43067f0892674a9cfbc6/src/transformers/models/reformer/modeling_reformer.py#L1685

Reply:
Resolved!